client/components/data/query-jetpack-modules/index.jsx

```diff
@@ -3,9 +3,12 @@ import { useEffect } from 'react';
 import { useDispatch } from 'react-redux';
 import { fetchModuleList } from 'calypso/state/jetpack/modules/actions';
 import isFetchingJetpackModules from 'calypso/state/selectors/is-fetching-jetpack-modules';
+import isSiteWpcomAtomic from 'calypso/state/selectors/is-site-wpcom-atomic';
 
 const request = ( siteId ) => ( dispatch, getState ) => {
-	if ( siteId && ! isFetchingJetpackModules( getState(), siteId ) ) {
+	const isAtomic = isSiteWpcomAtomic( getState(), siteId );
+
+	if ( siteId && ! isFetchingJetpackModules( getState(), siteId ) && isAtomic ) {
```

Comment:
It can be convenient to add this site type limitation here, but it also means we have to test every combination of the following:

* Area/component where the `QueryJetpackModules` query component is used
* Site type that supports (or doesn't support) Jetpack modules - Atomic sites, non-Atomic Jetpack sites, simple sites, etc.

Reply:
> Area/component where the QueryJetpackModules query component is used

I tested it on its usages, and they are working as expected:

- Pages -> All pages
- Settings -> Newsletter
- Tools -> Marketing

> Jetpack modules - Atomic sites, non-Atomic Jetpack sites, simple sites, etc.

- Atomic sites and non-Atomic Jetpack sites are working as expected [after this change](https://github.com/Automattic/wp-calypso/commit/2e34bb3a7e2ce8b3b0829a54ef4739f775a8f578).
- Simple site is not calling it as expected.